x-pack/plugins/maps/public/classes/styles/heatmap/heatmap_style.tsx

```diff
@@ -87,8 +87,11 @@ export class HeatmapStyle implements IStyle {
       radius = 8;
     }
     mbMap.setPaintProperty(layerId, 'heatmap-radius', radius);
-    const safeMax = max <= 0 ? 1 : max;
-    mbMap.setPaintProperty(layerId, 'heatmap-weight', ['/', ['get', propertyName], safeMax]);
+    if (max <= 0) {
```

Comment:
When I close the timeslider, there is a split second where `max = 1` which appears to cause this flash of red. There is a condition where the `layer._descriptor.__metaFromTiles` contains a single item where the properties are 
```
aggregations._count.avg: 1
aggregations._count.count: 1
aggregations._count.max: 1
aggregations._count.min: 1
aggregations._count.sum: 1
```

which causes `max = 1`. The loading indicator stops momentarily and then we see the flash of red before the loading indicator starts again. Let me know if you have any thoughts. 

https://user-images.githubusercontent.com/1638483/142953855-1a5b8091-76e0-44f4-ba3b-b8f74aa464c8.mp4

Reply:
Thanks. I can reproduce the issue and have pushed a fix.